core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -383,11 +412,30 @@ class LogManager(val logDirs: Array[File],
         cleaner.abortCleaning(topicAndPartition)
         cleaner.updateCheckpoints(removedLog.dir.getParentFile)
       }
-      removedLog.delete()
-      info("Deleted log for partition [%s,%d] in %s."
-           .format(topicAndPartition.topic,
-                   topicAndPartition.partition,
-                   removedLog.dir.getAbsolutePath))
+      // renaming the directory to topic-partition.unique-Id.delete
+      val dirName = StringBuilder.newBuilder
+      dirName.append(removedLog.name)
+      dirName.append(".")
+      val uniqueId = java.util.UUID.randomUUID.toString.replaceAll("-","")
```

Comment:
This is obviously not a big deal but I would have preferred to see a nice generation-like number instead of a UUID. If this makes the code unnecessarily complicated then never mind.